src/common/snippets/include/snippets/lowered/pass/extract_loop_invariants.hpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "pass.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+/**
+ * @interface ExtractLoopInvariants
+ * @brief Extract the exprs that produce identical result in loop iteration to outside of loop
+ * @ingroup snippets
+ */
+class ExtractLoopInvariants : public Pass {
```

Comment:
Can we make this pass `RanedPass` please? To have opportunity to launch on range of LinearIR

Reply:
sure, have done in [c575707](https://github.com/openvinotoolkit/openvino/pull/24044/commits/c5757078d0c1dbf4e73f824cb21c35a10b6d7cfd). Thanks Alexandra.